libs/frontend/feature-preview/src/lib/merge-search.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * searchB will take precedence over searchA.
+ */
+export const mergeSearch = (searchA: string, searchB: string): URLSearchParams => {
+  const urlA = new URL(`https://tmp${searchA}`)
```

Comment:
Is it necessary to use `tmp` here? I also wasn't sure about the `tmp` in `href-to-url.ts`

Reply:
The intention of `tmp` here is to make it a valid argument for the `URL` class. Anything that does not start with `http(s)://{host}` will lead to an error says "Uncaught TypeError: Failed to construct 'URL': Invalid URL".

Originally I tried to use `window.location.host` so that we don't have to use `tmp`, but then I realized that the test would fail in that case because `window` only exists in browser environment.

The use of `URL` class is to save us some time parsing the hash and query params ourselves. I should've left a comment here. Sorry about that!